src/terms/hubbard.jl

```diff
@@ -3,68 +3,62 @@ using Random
-    iatom_match   = isnothing(iatoms)  || (orbital.iatom in iatoms)
-    species_match = isnothing(species) || (species == orbital.species)
-    label_match   = isnothing(label)   || (label == orbital.label)
-    l_match = isnothing(l) || (l == orbital.l)
-    n_match = isnothing(n) || (n == orbital.n)
-    iatom_match && species_match && l_match && n_match && label_match
+function OrbitalManifold(psp::NormConservingPsp, iatoms::Vector{Int64}, label::AbstractString)
+    (; l, i) = find_pswfc(psp, label)
+    OrbitalManifold(psp, iatoms, l, i)
 end
-function is_on_manifold(orbital, manifold::OrbitalManifold)
-    is_on_manifold(orbital; iatoms=manifold.iatoms, l=manifold.l, n=manifold.i)
+function OrbitalManifold(atom::ElementPsp, iatoms::Vector{Int64}, label::AbstractString)
+    OrbitalManifold(atom.psp, iatoms, label)
 end
 
-function OrbitalManifold(atoms::Union{Vector{ElementPsp}, Vector{DFTK.Element}},
-                         labels; 
-                         iatoms::Union{Vector{Int64}, Nothing}=nothing,
-                         label::Union{String, Nothing}=nothing,
-                         species::Union{Symbol, AtomsBase.ChemicalSpecies, Nothing}=nothing)
-    hub_atoms = Int64[]
-    for orbital in labels
-        if is_on_manifold(orbital; iatoms, species, label) 
-            append!(hub_atoms, orbital.iatom)
+function find_pswfc(psp::NormConservingPsp, label::String)
+    for l = 0:psp.lmax, i = 1:count_n_pswfc_radial(psp, l)
+        if pswfc_label(psp, i, l) == label
+            return (; l, i)
         end
     end
-    isempty(hub_atoms) && error("Unable to create Hubbard manifold. No atom matches the given keywords")
-    # If species is nothing, there can be errors if the iatoms correspond to different atomic species
-    model_atom = atoms[hub_atoms[1]]
-    !all(atom -> atom.psp == model_atom.psp, atoms[hub_atoms]) && 
-        error("The given Hubbard manifold contains more than one atomic pseudopotential species")
-    l = isnothing(label) ? nothing : labels[hub_atoms[1]].l
-    i = isnothing(label) ? nothing : labels[hub_atoms[1]].n
-    OrbitalManifold(;iatoms=hub_atoms, psp=model_atom.psp, l, i)
+    error("Could not find pseudo atomic orbital with label $label "
+          * "in pseudopotential $(psp.identifier).")
+end
+
+function check_hubbard_manifold(manifold::OrbitalManifold, model::Model)
+    for atom in model.atoms[manifold.iatoms]
+        atom isa ElementPsp || error("Orbital manifold elements must have a psp.")
+        atom.psp === manifold.psp || error("Orbital manifold psp does not match the psp of atom $atom")
+    end
+    isempty(manifold.iatoms) && error("Orbital manifold has no atoms.")
+    # Tricky: make sure that the iatoms are consistent with the symmetries of the model,
+    #         i.e. that the manifold would be a valid atom group
+    _check_symmetries(model.symmetries, model.lattice, [manifold.iatoms], model.positions)
```

Comment:
I don't think people will understand the error message that comes out of this, if it triggers (function has originally been written for sth else), but certainly should be checked explicitly.